extensions/wikia/AdEngine/js/PageFairDetection.js

```diff
@@ -0,0 +1,73 @@
+define('ext.wikia.adEngine.pageFairDetection', [
+	'ext.wikia.adEngine.adContext',
+	'ext.wikia.adEngine.utils.scriptLoader',
+	'wikia.document',
+	'wikia.log',
+	'wikia.window'
+], function (adContext, scriptLoader, doc, log, window) {
+	'use strict';
+
+	var context = null,
```

Comment:
When the module is initialized it's too soon to run function `getContext` because part of it is not loaded yet. If this way isn't clear enough IMO the best way will be your solution with injecting context by `init` function